include/bluetooth/services/hids_c.h

```diff
@@ -0,0 +1,599 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef BT_GATT_HIDS_C_H_
+#define BT_GATT_HIDS_C_H_
+
+/**
+ * @file
+ * @defgroup bt_gatt_hids_c BLE GATT HIDS Client API
+ * @{
+ * @brief API for the BLE GATT HID Service (HIDS) Client.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <bluetooth/gatt.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/common/gatt_dm.h>
+
+struct bt_gatt_hids_c;
+struct bt_gatt_hids_c_rep_info;
+
+/**
+ * @brief Protocol mode
+ *
+ * Possible values for protocol mode characteristic value
+ */
+enum bt_gatt_hids_c_pm {
+	BT_GATT_HIDS_C_PM_BOOT   = 0x00,/**< Boot protocol   */
+	BT_GATT_HIDS_C_PM_REPORT = 0x1  /**< Report protocol */
+};
+
+/**
+ * @brief Report types
+ *
+ * Report types as as defined in Report Reference characteristic descriptor.
+ */
+enum bt_gatt_hids_c_report_type {
+	BT_GATT_HIDS_C_REPORT_TYPE_RESERVED = 0x00, /**< Reserved value */
+	BT_GATT_HIDS_C_REPORT_TYPE_INPUT    = 0x01, /**< Input Report   */
+	BT_GATT_HIDS_C_REPORT_TYPE_OUTPUT   = 0x02, /**< Output Report  */
+	BT_GATT_HIDS_C_REPORT_TYPE_FEATURE  = 0x03  /**< Feature Report */
+};
+
+/**
+ * @brief Callback function when notification or read response is received
+ *
+ * Function is called when new data related to the given report object
+ * appears.
+ * The data size can be obtained from the report object.
+ *
+ * @param hids_c HIDS client object
+ * @param rep    Report object
+ * @param err    ATT error code
+ * @param data   Pointer to the received data
+ *
+ * @retval BT_GATT_ITER_STOP     Stop notification
+ * @retval BT_GATT_ITER_CONTINUE Continue notification
+ */
+typedef u8_t (*bt_gatt_hids_c_read_cb)(struct bt_gatt_hids_c *hids_c,
+				       struct bt_gatt_hids_c_rep_info *rep,
+				       u8_t err,
+				       const u8_t *data);
```

Comment:
You mention that `data size can be obtained from the report object`. How this can be done?

Reply:
Comment added.